source/constructs/api/catalog/service.py

```diff
@@ -736,14 +737,28 @@ def sync_job_detection_result(
         overwrite=True
 ):
     logger.info("start time")
+    deel_s3_all = True
+    if database_type == DatabaseType.S3.value:
+        job_run = job_crud.get_run(job_run_id)
+        if not job_run:
+            raise Exception("Query job error")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.